plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineExecutionService.java

```diff
@@ -43,7 +43,7 @@ public PipelineExecutionService(PipelineStore store, ThreadPool threadPool) {
         this.threadPool = threadPool;
     }
 
-    public void execute(IndexRequest indexRequest, String pipelineId, ActionListener<IngestDocument> listener) {
+    public void execute(IndexRequest indexRequest, String pipelineId, ActionListener<Void> listener) {
```

Comment:
that is a good reason to remove the ingest document itself from the action listener.